lib/templateFactory.js

```diff
@@ -264,46 +265,60 @@ class TemplateFactory extends BaseFactory {
 
             const templateIds = templates.map(t => t.id);
 
-            const metricJobsListConfig = {
-                params: { templateId: templateIds },
-                readOnly: true,
-                aggregationField: 'templateId'
-            };
-
-            const metricBuildsListConfig = { ...metricJobsListConfig };
+            // first need to get the job ids that are not archived for the templates
+            return jobFactory
+                .list({
+                    params: { templateId: templateIds, archived: false },
+                    readOnly: true,
+                    aggregationField: 'id'
+                })
```

Comment:
We can skip this step.
Instead we could just add `archived: false` filter wherever needed.

Reply:
We need not filter out builds that are associated with archived jobs.

Reply:
Is there another way to do that using the current `buildFactory` model apart from passing the job ids to filter the builds belong to archived jobs out? When I looked at the [data-schema for the build](https://github.com/screwdriver-cd/data-schema/blob/master/models/build.js#L27-L104), `jobId` was the first came to mind. Opened to input.

Reply:
We don't need to filter builds based on whether the latest configuration of the associated job no longer uses that template or uses a different template or the job is archived.

This PR should only fix pipeline usage and job usage metrics. There is no change needed for build usage metric.